action.yaml

```diff
@@ -130,51 +128,42 @@ runs:
-          "${GITHUB_WORKSPACE}/citadel-config.yaml" "${{ env.PROJECT_PATH }}" "$GITHUB_WORKFLOW_REF" "$RUNNER_TEMP"
-        
-        echo "CITADEL_JSON_PATH=$RUNNER_TEMP/citadel.json" >> $GITHUB_ENV
-
     - name: Upload to the citadel host
       if: ${{ env.CITADEL_JSON_PATH }}
       id: citadel 
       shell: bash
-      run: |        
-        file_paths="${{ env.CITADEL_JSON_PATH }}"
-        # unset CITADEL_JSON_PATH so that it doesn't get trigger again by other steps
-        echo "CITADEL_JSON_PATH=" >> $GITHUB_ENV
-        IFS=', ' read -ra file_paths_array <<< "$file_paths"
-        for file_path in "${file_paths_array[@]}"; do 
-          echo "Processing File $file_path"
-          cat $file_path
-          curl -vvvX POST '${{ env.CITADEL_URL }}' --silent --fail --show-error -d @$file_path -H 'Content-Type: application/json' -H "Connection: close"  
+      run: |
+        VALUES="${{ env.CITADEL_JSON_PATH }}"  
+        IFS=', ' read -ra VALUES_ARRAY <<< "$VALUES"
+        for VALUE in "${VALUES_ARRAY[@]}"; do
+          echo                               #Empty Line 
+          echo "Processing File $VALUE"
+          curl -vX POST 'http://${{ env.citadel-host }}:5000/citadel' -d ${{ github.workspace }}/$VALUE --header 'Content-Type: application/json'
           sleep ${{ env.CITADEL_JSON_LIST_SLEEP_SECONDS || 60  }}   # If env not passed, Sleeping for 60 seconds (default)
         done
-        
-    # Blackduck section
+
+    # Blackduck section    
     - name: Blackduck Scan
-      if: ${{ env.BlackDuck_Project_Name }}
+      if: ${{ always() && env.BlackDuck_Project_Name }}
       id: blackduck-scan
       uses: addnab/docker-run-action@v3
       with:
         image: docker.repo.orl.eng.hitachivantara.com/blackducksoftware/detect:8
-        options: --entrypoint "/bin/bash" -v ${{ github.workspace }}:/workdir
+        options: --entrypoint "/bin/bash" -v ${{ github.workspace }}:/workdir 
         run: |
-          java -jar /synopsys-detect.jar \
+          java -jar /synopsys-detect.jar  \
            --detect.source.path=${{ env.BlackDuck_Source_Path || '/workdir' }} \
            --detect.project.version.name="${{ env.BlackDuck_Project_Version }}" \
            --detect.project.name="${{ env.BlackDuck_Project_Name }}" \
            --detect.accuracy.required=NONE \
            --blackduck.api.token="${{ env.BlackDuck_Api_Token }}" \
            --blackduck.url="${{ env.BlackDuck_Url }}" \
-           --detect.blackduck.signature.scanner.snippet.matching="NONE" ${{ env.ADDITIONAL_ARGS }}
-
+           --detect.blackduck.signature.scanner.snippet.matching="NONE" ${{ env.ADDITIONAL_ARGS }} > $RUNNER_TEMP/detector.log 2>&1
+          cat $RUNNER_TEMP/detector.log
+          blackduck_url=$(grep -i "Black Duck Project BOM" $RUNNER_TEMP/detector.log)
+          blackduck_url=$(echo "$blackduck_url" | grep -o 'https://[^ ]*' | cut -d ' ' -f1)
+          echo "url=$blackduck_url" > ./workdir/blackduck_url.txt
```

Comment:
If you add files to the workspace, they might be committed unwilling. I don't understand... you're creating the `detector.log` inside `$RUNNER_TEMP`... is that path available inside the container?! If so why not just create `blackduck_url.txt` in `$RUNNER_TEMP` also?

Reply:
Here when we are creating blackduck_url.txt in $RUNNER_TEMP we are not able to read the file in next step because it is creating the file inside the blackduck container that's why we are creating the blackduck_url.txt in workdir.